class.jetpack.php

```diff
@@ -5395,6 +5422,18 @@ function wp_rest_authenticate( $user ) {
 		return null;
 	}
 
+	protected function send_signature_error_header( $error ) {
+		$error_data = $error->get_error_data();
+		if ( ! isset( $error_data['signature_details'] ) ) {
+			return;
+		}
+
+		header( sprintf(
+			'X-Jetpack-Signature-Error: %s',
+			base64_encode( json_encode( $error_data['signature_details'] ) )
+		) );
```

Comment:
Can you simplify the logic here so the function doesn't return void on one case and nothing on the other?
```
		if ( isset( $error_data['signature_details'] ) ) {
			header( sprintf(
				'X-Jetpack-Signature-Error: %s',
				base64_encode( json_encode( $error_data['signature_details'] ) )
			) );
		}
```

Reply:
Why does it matter what this function returns? We are not checking the output anywhere.

Nested if's don't really simplify the logic IMO, they just move it around.

Reply:
There is no practical difference between `return;` and not calling `return`. Both return `null`:

```php
function return_nothing() {
  return;
}

function do_not_return() {
   // nothing
}

function return_null() {
  return null;
}

var_dump( return_nothing(), do_not_return(), return_null() ); // NULL, NULL, NULL
```

The only difference between `void` and `null` occurs when using `void` as a return type hint.

```php
function this_works() : void {
  return; // fine
}

function this_also_works() : void {
   // nothing - also fine
}

function this_breaks() : void {
  return null; // not fine
}
```

The last function will cause a parse error.

So PHP knows the difference, but calling code can't. (Well - there might be a way to tell the difference with reflection, but that's silly :))

In the case of this `send_signature_error_header()` method, if we were to include a `void` return type hint, everything would be fine, so my claim is that this falls into the "there is no practical difference" category :)

I think an aesthetic/code style argument could be made, but I (subjectively) disagree ;) I prefer the "bail as early as possible and indent as little as possible" argument in this case.

Reply:
(Moot now because the code is gone ;))